code/modules/mob/living/simple_animal/friendly/mushroom.dm

```diff
@@ -23,6 +25,14 @@
 /mob/living/simple_animal/mushroom/New()
 	..()
 	harvest_time = world.time
+	var/count = 0
+	for (var/mob/living/simple_animal/mushroom in world)

```

Comment:
Check the place that has all of the global lists for a list of all mobs. It'll be much faster to search that, as opposed to going through every object in the world.